src/index.js

```diff
@@ -90,11 +90,12 @@ class Plugin {
       insufficientDataActions.push(alertTopics.insufficientData);
     }
 
+    const stackName = this.awsProvider.naming.getStackName();
     const namespace = definition.pattern ?
-      this.awsProvider.naming.getStackName() :
+      stackName :
       definition.namespace;
 
-    const metricName = definition.pattern ?
+    const metricLogicalId = definition.pattern ?
```

Comment:
This is not actually the logicalId, is it? The logicalId is a CloudFormation concept, and is the [key] of the Resources object in CF template.  This MetricName is just a reference to AWS, where the Metric could exist outside of the stack, maybe?

Reply:
You are right, but it is different from basic metric name for pattern-based alarms. How would you name it then?

Reply:
maybe just metricId ? the 'logical' part feels like a specific cloudformation detail?

Reply:
hmm.. makes sense.